File: go-client-sdk/pkg/models/operations/postcouponscouponidarchive.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
)

type PostCouponsCouponIDArchivePathParams struct {
	CouponID string `pathParam:"style=simple,explode=false,name=coupon_id"`
}

type PostCouponsCouponIDArchiveRequest struct {
	PathParams PostCouponsCouponIDArchivePathParams
}

type PostCouponsCouponIDArchiveResponse struct {
	ContentType string
	Coupon      *shared.Coupon
	StatusCode  int
}

File: typescript-client-sdk/src/sdk/models/operations/postcouponscouponidarchive.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class PostCouponsCouponIdArchivePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=coupon_id" })
  couponId: string;
}

export class PostCouponsCouponIdArchiveRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PostCouponsCouponIdArchivePathParams;
}

export class PostCouponsCouponIdArchiveResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  coupon?: shared.Coupon;

  @SpeakeasyMetadata()
  statusCode: number;
}